src/merkle.rs

```diff
@@ -237,19 +624,106 @@ impl<T: Element, A: Algorithm<T>, K: Store<T>, U: Unsigned> MerkleTree<T, A, K,
         );
 
         Ok(MerkleTree {
-            data,
+            data: Some(data),
             leafs,
+            len: tree_len,
             height,
+            top_layer_nodes: 0,
+            trees: None,
+            sub_tree_layer_nodes: 0,
+            sub_trees: None,
             root,
             _u: PhantomData,
             _a: PhantomData,
             _t: PhantomData,
+            _n: PhantomData,
+            _r: PhantomData,
         })
     }
 
+    /// Generate merkle sub tree inclusion proof for leaf `i` for
+    /// either the top layer or the sub-tree layer, specified by the
+    /// top_layer flag
+    #[inline]
+    fn gen_sub_tree_proof(
+        &self,
+        i: usize,
+        top_layer: bool,
+        arity: usize,
```

Comment:
The arity could be a type argument, as far as I can tell from the usage of this method

Reply:
@cryptonemo did you forget about this or decide against it?